### PR TITLE
[FLINK-33303] [flink-avro] Union type not inheriting type conversions

### PR DESCRIPTION
**Pull request by m8719-github**
## What is the purpose of the change

Apache Avro 1.11.3 contains an important fix: [AVRO-3536](https://issues.apache.org/jira/browse/AVRO-3536)
Just want to make sure it makes it into the next Flink release.

## Brief change log

Bumped up Apache Avro version to 1.11.3

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **yes**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**

### CODE REVIEW

**Comment by afedulov** [comment]
@m8719-github thanks for the contribution. All Flink PRs need to have a corresponding issue in Flink JIRA. Could you please create one with a brief description of the issue being addressed?
https://issues.apache.org/jira/projects/FLINK/issues/

**Comment by afedulov** [comment]
Actually, there is already an earlier PR with this bump to address vulnarabilities:
https://github.com/apache/flink/pull/23508

File: flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/RegistryAvroDeserializationSchemaTest.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ void testSpecificRecordReadMoreFieldsThanWereWritten() throws IOException {
                                         + " \"fields\": [\n"
                                         + "     {\"name\": \"name\", \"type\": \"string\"}"
                                         + " ]\n"
-                                        + "}]");
+                                        + "}");
```
[the remark this answers is not in the document]
Comment (m8719-github) [reply flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/RegistryAvroDeserializationSchemaTest.java:88]: yes, looks like it.

**Comment by m8719-github** [comment]
I've reached out to Martin here: [FLINK-33238](https://issues.apache.org/jira/browse/FLINK-33238) to see if we can consolidate.

**Comment by m8719-github** [comment]
@afedulov Created and [FLINK-33303](https://issues.apache.org/jira/browse/FLINK-33303)
Do you think there is still time to get this into Flink 1.18 ?

**Comment by afedulov** [comment]
> Do you think there is still time to get this into Flink 1.18 ?

Not unless it fixes some critical vulnerability or test regression. 1.18 was supposed to be released end of September, so I am afraid we are way beyond the features integration timeframe.
https://cwiki.apache.org/confluence/display/FLINK/1.18+Release

**Comment by MartijnVisser** [comment]
Fixed by https://github.com/apache/flink/pull/23508 - Will be backported to 1.18 (so for 1.18.1) with https://github.com/apache/flink/pull/23559

**Pull request closed without merging**